model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/RealmCacheSession.java

```diff
@@ -620,31 +622,30 @@ public RoleModel addRealmRole(RealmModel realm, String id, String name) {
             Long loaded = cache.getCurrentRevision(cacheKey);
             Set<RoleModel> model = getRealmDelegate().getRealmRoles(realm);
```

Comment:
We discussed this with @hmlnarik and @sguilhen and we agreed on this solution. Problem here is the cashing of role ids. If you obtain a stream here, it will be spend by transformation to Stream<String> (for cashing) and therefore the stream couldn't be returned later.

Reply:
This is an excellent question.

Generally, this is the case. In this case, it is however justified, as the cache provider is the final recipient of all the roles that it needs to cache, and it needs to process all of them, so it would collect them anyway.

Later, we can think of optimizations like lazy-loading of the role details, and only returning stubs with IDs that would load the details only after when they are accessed. This would help across multiple places in caching.

Reply:
So maybe we shouldn't deprecated the method(s) after all. As far as I understand there are cases when using non-stream method is actually more efficient, right?

Reply:
So after discussion with @hmlnarik we agreed we should add comment something like `// this method is use intentionally here because ...` to places where the deprecated methods are used.

Reply:
comments added